app/controllers/dashboard/internship_offer_infos_controller.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Dashboard
+  class InternshipOfferInfosController < ApplicationController
+    before_action :authenticate_user!
+
+    def new
+      @internship_offer_info = InternshipOfferInfo.build_from_internship_offer(InternshipOffer.find(params[:duplicate_id])) if params[:duplicate_id].present?
+      @internship_offer_info ||= InternshipOfferInfo.new
+      authorize! :create, InternshipOfferInfo
+      @available_weeks = Week.selectable_from_now_until_end_of_school_year
+      @organisation_id = params[:organisation_id]
+    end
+
+    def create
+      @internship_offer_info = InternshipOfferInfo.new(internship_offer_info_params.merge!(prepare_daily_hours(params)))
+      organisation_id = params[:internship_offer_info][:organisation_id]
+
+      if @internship_offer_info.save
+        redirect_to new_dashboard_internship_offer_path(
+          organisation_id: organisation_id,
+          internship_offer_info_id: @internship_offer_info.id,
+        )
+      else
+        @available_weeks = Week.selectable_from_now_until_end_of_school_year
+        @organisation_id = organisation_id
+        render :new, status: :bad_request
+      end
+    end
+
+    def edit
+      @internship_offer_info = InternshipOfferInfo.find(params[:id])
+      authorize! :edit, InternshipOfferInfo
```

Comment:
@pierrax wrong ACL, edition should check on instance (owned by current user only)